src/hooks/useUpdateCountries.ts

```diff
@@ -1,23 +1,15 @@
-import { useMutation, useQueryClient } from 'react-query';
+import { useMutation, useQueryClient } from "react-query";
 
 export default function useUpdateCountries() {
   const queryClient = useQueryClient();
 
-  return useMutation(
-    (value?: any) => {
-      const { regionName, regions, countries } = value;
-      const region = regions.find((item) => item.display_value === regionName);
-      let newList = countries;
-      if (region) {
-        newList = countries.filter((item: any) => item.parent_id === region.id);
-      }
-      return queryClient.setQueryData('filteredCountries', (old) => newList);
-    },
-
-    {
-      onError: (err) => {
-        console.log(err);
-      },
+  return useMutation(async (value?: any) => {
+    const { regionName, regions, countries } = value;
+    const region = regions.find((item) => item.slug === regionName);
+    let newList = countries;
+    if (region) {
+      newList = countries.filter((item: any) => item.parent_id === region.id);
```

Comment:
Notice we're missing the `onError` is that deliberate?

Reply:
As these are 'internal' mutations to update the local state I don't see the need to console log the errors. I agree we will need to add some better error handling for API communication though.